### PR TITLE
Added spectate mode.

### LINKED ISSUE #7: Support spectate mode

### PR DESCRIPTION
**Pull request by henopied**
Closes #7.

### CODE REVIEW

File: client/js/pointer.js [hunk 1]
```diff
@@ -19,6 +19,19 @@ class Pointer {
         this.client.eject();
       } else if (e.keyCode == 32) {
         this.client.split();
+      } else if (e.keyCode == 81) {
+        let freeMoveBuf  = new ArrayBuffer(1);
+        let freeMoveView = new DataView(freeMoveBuf);
+        freeMoveView.setUint8(0, 18);
+        this.client.send(freeMoveView.buffer);
```
Comment (heroandtn3) [inline client/js/pointer.js]: The agario-client already supports this:

> client.spectateModeToggle() switching spectate mode in spectating mode (Q key in official client). Use client.moveTo() to move your "camera" around. Look at client.on('spectateFieldUpdate') for movement updates. Will return false if connection is not established.

**Comment by henopied** [comment]
I did notice that but when I had tried it out before it did not seem to work properly and this also closer matches agar.io's implementation.

**Comment by heroandtn3** [comment]
Actually, we copied agario-client to our code base, you can fix it if it doesn't work ;). Here is the function: https://github.com/SNSA/NEXT/blob/master/client/js/agario-client/client.js#L588

**Comment by henopied** [comment]
I think it had to do with the fact those 2 packets were being sent at the same time. I suppose I could add a setTimeout to the second packet.

**Comment by heroandtn3** [comment]
I just made a test spectate mode, the game was very lag and CPU usage was always around 100%. Do you get the same?

**Comment by henopied** [comment]
Yes it is extremely laggy that's why I made those issues about zsort and destroying the balls.

**Comment by heroandtn3** [comment]
Thanks for your update, could you please squash your 4 commit to 1?

**Comment by henopied** [comment]
Yes, I would do that but for some reason I am having trouble using squash might you be able to show me a good guide on how to use it?

**Comment by heroandtn3** [comment]
Sure, I'm on Gitter https://gitter.im/SNSA/NEXT

**Pull request merged**